src/utils/common.ts

```diff
@@ -129,10 +137,133 @@ export const dirtyFieldsToRequestObject = (dirtyFields: object, form: IAppForms)
 
   syncPlanningDates(request, form);
   syncConstructionDates(request, form);
+  syncFinances(request, project);
 
   return request;
 };
 
+/**
+ * Make sure the finances are moved to correct years if dates are changed
+ * @param project IProject being updated
+ * @param data IProjectRequest object with data to be patched
+ */
+export const syncFinances = (request: IProjectRequest, project: IProject | null) => {
+  if (!project?.finances) {
+    return;
+  }
+
+  const adjustBudgets = (
+    budgetObject: IProjectFinances,
+    projectTimelineDates: IProjectEstDates,
+  ): IProjectFinances => {
+    /**
+     * Moves the provided finance value to its new finance location and sets current to null
+     * @param newFieldYear year for the new finance field which needs to be updated
+     * @param financeValue value which will be summed in the new year
+     * @param currentFinanceFieldName finance field name which the value belongs to
+     */
+    const updateFinanceValue = (
```

Comment:
Move this function outside the adjustBudgets function and provider parameter for adjustedBudgetObjet. Currently it is very confusing implementation